rust/gitxetcore/src/git_integration/git_repo.rs

```diff
@@ -230,6 +235,65 @@ impl GitRepo {
         Self::open_impl(config, true)
     }
 
+    // Create a list of remote URLs to try to query to find the upstream xet remote based on the existing
+    // formats of the upstream remote stuff.
+    fn get_xet_upstream_remote_urls(&self, upstream_info: &UpstreamXetRepo) -> Result<Vec<String>> {
+        let mut ret = Vec::new();
+
+        let Some(origin_type) = upstream_info.origin_type.as_ref() else {
+            return Ok(vec![]);
+        };
+
+        if origin_type == "github" {
+            if let Some(url) = &upstream_info.url {
+                ret.push(url.clone());
+            }
+
+            if let (Some(user_name), Some(repo_name)) =
+                (&upstream_info.user_name, &upstream_info.repo_name)
+            {
+                // Now, figure out whether we're using https or git as the protocol.  To do this,
+                // we scan through the origin remotes to determine which could be an upstream git remote.
+                // This will tell us whether to form it as an ssh url or an https url.
+
+                // First, determine the origin name.
+                let (ssh_is_first, ssh_only) = 'b: {
+                    // Try "origin" first, then try the rest of the remotes in order to find a github remote.
+                    for r_name in ["origin"]
+                        .into_iter()
+                        .chain(self.repo.remotes()?.iter().flatten())
+                    {
+                        if let Ok(r) = self.repo.find_remote(r_name) {
+                            let url = r.url().unwrap_or_default();
+                            if url.contains("github") {
+                                break 'b (url.contains("git@github"), true);
```

Comment:
Not sure if we really need the `ssh_only` flag. It's a guess anyway?

Reply:
If the authentication for the https is wrong, then it hangs asking for user input...

Reply:
Ah, that would be annoying. Make sense to me!